lib/logstash/outputs/scalyr.rb

```diff
@@ -619,29 +641,32 @@ def build_multi_event_request_array(logstash_events)
         # make sure we always have at least one event
         if scalyr_events.size == 0
           scalyr_events << scalyr_event
+          l_events << l_event
           append_event = false
         end
-        multi_event_request = self.create_multi_event_request(scalyr_events, current_threads, logs)
+        multi_event_request = self.create_multi_event_request(scalyr_events, l_events, current_threads, logs)
         multi_event_request_array << multi_event_request
 
         total_bytes = 0
         current_threads = Hash.new
         logs = Hash.new
         logs_ids = Hash.new
         scalyr_events = Array.new
+        l_events = Array.new
```

Comment:
Hm, this could be problematic for large batches since it means memory consumption will now basically double, at least as far as batch which is currently being processed goes, right?

Reply:
Which may be OK and not add to that much, but we just need keep an eye on it.

Reply:
And on that note - before we have a better idea on how much overhead it introduces we should probably release a beta version of leave it disabled by default.

Reply:
Good point, I think I can rework this to just keep track of the index range of events, which we can then grab from the original array.

Reply:
Though having said that, this array should be holding references to the events already in that array and not full new objects, so not actually much more memory use. Unless I'm misunderstanding how Ruby handles objects.

Reply:
Yeah if you could refactor it to just use the offsets, that would be great.

EDIT: If it's just holding a reference to the event hash object then, yeah, it should probably be fine.

Reply:
From a bit of research it seems to be the case that its only references, so there shouldn't be a noticeable increase in memory use.

Reply:
cool, sgtm, but let's still do beta release first once this PR is merged.